powerauth-nextstep/src/main/java/io/getlime/security/powerauth/app/nextstep/service/UserIdentityService.java

```diff
@@ -642,13 +644,26 @@ private void removeAllCredentials(UserIdentityEntity user) {
      * @param user User identity entity.
      */
     private void removeAllOtps(UserIdentityEntity user) {
-        final List<OtpEntity> otps = otpRepository.findAllByUserId(user.getUserId());
-        otps.forEach(otp -> {
-            if (otp.getStatus() != OtpStatus.REMOVED) {
+        List<OtpEntity> removedOtps = new ArrayList<>();
+        try (final Stream<OtpEntity> activeOtps = otpRepository.findAllByUserIdAndStatus(user.getUserId(), OtpStatus.ACTIVE)) {
```

Comment:
Maybe again a task for the Update query instead of the Select via Stream?

Reply:
Ah, yes, that would be better.